lib/wings/services/file_metadata_builder.rb

```diff
@@ -39,18 +39,30 @@ def attach_file_metadata(file_metadata:, file_set:)
 
     private
 
+      ##
+      # @api private
       def attach_file_metadata_to_valkyrie_file_set(file_metadata, file_set)
         # This is for storage adapters other than wings.  The wings storage adapter already attached the file to the file_set.
         # This process is a no-op for wings.  # TODO: WINGS - May need to verify this is a no-op for wings once file_set is passed in as a resource.
         # TODO: WINGS - Need to test this against other adapters once they are available for use.
-        existing_file_metadata = file_set.original_file || file_metadata
+        existing_file_metadata = current_original_file(file_set) || file_metadata
         file_metadata = existing_file_metadata.new(file_metadata.to_h.except(:id))
         saved_file_metadata = persister.save(resource: file_metadata)
         file_set.file_ids = [saved_file_metadata.id]
         persister.save(resource: file_set)
         saved_file_metadata
       end
 
+      ##
+      # @api private
+      # @return [Hyrax::FileMetadata, nil]
+      def current_original_file(file_set)
+        Hyrax.query_service
+          .custom_queries
+          .find_many_file_metadata_by_use(resource: file_set, use: Hyrax::FileMetadata::Use::ORIGINAL_FILE)
```

Comment:
Part of what I liked about the #original_file method was that it was simple and had the knowledge that there should be only one, so it was ok to grab the first one returned.  I'd hate to see this code block repeated in multiple places.  If it doesn't belong in FileSet (for single responsibility reasons), is there somewhere else it can live (to keep it DRY)?  And if not, should it be a service.  Getting the original file is something that happens a lot in Hyrax.

Reply:
~~I think the idea of a service is a good one. Maybe there should be a navigator?~~
There already is one. I'll look to make sure it works well and refactor around it.

Reply:
can we hold this for a later refactor? i promise to get to it this week, if so.

Reply:
I am good with putting this in another PR.  Created an issue to track this...  #4233